packages/babel-traverse/src/scope/index.js

```diff
@@ -761,7 +788,10 @@ export default class Scope {
   }
 
   init() {
-    if (!this.references) this.crawl();
```

Comment:
this check was super implicit and confusing (nested scopes don't even gather `references` so this object stays always empty for them, but is initialized in `.crawl`)